fdbrpc/FlowTransport.actor.cpp

```diff
@@ -727,14 +730,14 @@ ACTOR static Future<Void> connectionReader(
 					// At the beginning of a connection, we expect to receive a packet containing the protocol version and the listening port of the remote process
 					int32_t connectPacketSize = ((ConnectPacket*)unprocessed_begin)->totalPacketSize();
 					if ( unprocessed_end-unprocessed_begin >= connectPacketSize ) {
-						uint64_t protocolVersion = ((ConnectPacket*)unprocessed_begin)->protocolVersion;
+						auto protocolVersion = ((ConnectPacket*)unprocessed_begin)->protocolVersion;
 						BinaryReader pktReader(unprocessed_begin, connectPacketSize, AssumeVersion(protocolVersion));
 						ConnectPacket pkt;
 						serializer(pktReader, pkt);
 
 						uint64_t connectionId = pkt.connectionId;
-						if(g_network->useObjectSerializer() != hasObjectSerializerFlag(pkt.protocolVersion) ||
-						   (removeFlags(pkt.protocolVersion) & compatibleProtocolVersionMask) != (currentProtocolVersion & compatibleProtocolVersionMask)) {
+						if(g_network->useObjectSerializer() != pkt.protocolVersion.hasObjectSerializerFlag() ||
```

Comment:
I wonder if we still want to have the same notion of 'protocol version compatibility' even with the object serializer, in case we want to make some radical change at the expense of forward compatibility. I'm imagining that the value of the object serializer is that it's much easier to make changes that are protocol-compatible.

This could probably use a design doc. I can write a draft if you like

Reply:
Yes, I think this will need to change - but this is out of scope for this PR - we need to talk about this and make another change later (after we have the flatbuffers-optimizations in FDB 6)